Test/dafny0/Compilation.dfy

```diff
@@ -234,7 +234,7 @@ module GhostLetExpr {
         ghost var z := bb + F();
         ghost var t0 := var y := z; z + 3;
         ghost var t1 := ghost var y := z; z + 3;
-        var t2 := ghost var y := z; aa + 3;
+        var t2; t2 := ghost var y := z; aa + 3;
```

Comment:
Isn't this still okay as it was? Under the new rules, when `t2` is declared with an initializing expression, it is auto-ghost. Being auto-ghost means `t2` is automatically declared `ghost` if the RHS of the initializing assignment is ghost. But the expression `ghost var y := z; a + 33` is not ghost. So, I would expect that the auto-ghost `t2` will still be non-ghost. Am I understanding this correctly?

Reply:
var t2; declares t2 to be non-ghost. So writing var t2; t2 := ghost var y := z; aa + 3; checks that aa+3 is non-ghost. Writing var t2 := ghost var y := z; aa + 3; delays any incorrect error message to a use of t2 in a print statement (t2 is auto-ghost and ends up being non-ghost).

Reply:
Added test cases and implementation code for the additional example